### PR TITLE
Update deps to use polkadot sdk, unfork nomination pool, remove unused padding and introduce header V3

### PR DESCRIPTION
**Pull request by Leouarz**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- [x] Feature
- [x] Refactor

## Description

- We now use Polkadot-sdk as deps
- It contains an upgrade to polkadot v1.1
- It unfork the nomination pool pallet since the bug we correct was correct in upper versions (so no need to have it locally)
- Removes the padding_tail_value added for each unique app id in a block matrix and tests are updated
- A new header v3 was added for backward compatibility for padding_tail_value

You can view PRs separately here :
- https://github.com/availproject/avail/pull/365
- https://github.com/availproject/avail/pull/371

## Checklist
- [x] I have performed a self-review of my own code.
- [x] The tests pass successfully with `cargo test`.
- [x] The code was formatted with `cargo fmt`.
- [x] The code compiles with no new warnings with `cargo build --release` and `cargo build --release --features runtime-benchmarks`.
- [x] The code has no new warnings when using `cargo clippy`.
- [x] If this change affects documented features or needs new documentation, I have created a PR with a [documentation update](https://github.com/availproject/availproject.github.io).
- [x]  Synced network with full and warp
- [x] Binary upgrade then Runtime upgrade were successfull

### CODE REVIEW

File: e2e/src/main.rs [hunk 1]
```diff
@@ -186,7 +187,15 @@ pub mod tests {
 		let app_extrinsics = get_block_app_extrinsics(&submitted_block).unwrap();
 
 		// Grid Creation
-		let grid = EvaluationGrid::from_extrinsics(app_extrinsics, 4, 256, 256, [0u8; 32]).unwrap();
+		let grid = EvaluationGrid::from_extrinsics(
```
Comment (markopoloparadox) [inline e2e/src/main.rs:190]: nooo, it got ugly :(

File: pallets/system/src/header_builder.rs [hunk 2]
```diff
@@ -1,26 +1,18 @@
...
 use sp_runtime::SaturatedConversion;
 use sp_runtime::{generic::Digest, traits::Hash};
-use sp_runtime_interface::{pass_by::PassByCodec, runtime_interface};
+use sp_runtime_interface::runtime_interface;
 use sp_std::vec::Vec;
 
 use crate::{limits::BlockLength, Config, LOG_TARGET};
 
-#[derive(Debug, Clone, Copy, Eq, PartialEq, PassByCodec, Encode, Decode, TypeInfo)]
-pub enum HeaderVersion {
```
Comment (markopoloparadox) [inline pallets/system/src/header_builder.rs:19]: RIP

File: runtime/src/constants.rs [hunk 3]
```diff
@@ -277,6 +278,13 @@ pub mod staking {
 			.max
 			.get(DispatchClass::Normal);
 		pub const OffchainRepeat: BlockNumber = 5;
+
+		// Note: the EPM in this runtime runs the election on-chain. The election bounds must be
```
Comment (markopoloparadox) [inline runtime/src/constants.rs:282]: Does this mean anything for us?

**Review by markopoloparadox — APPROVED** [review]

File: e2e/src/main.rs (hunk 1 above)
Comment (Leouarz) [reply e2e/src/main.rs:190]: Sadge header version noises 🔊

File: runtime/src/constants.rs (hunk 3 above)
Comment (Leouarz) [reply runtime/src/constants.rs:282]: I think it applies for us even if we won't reach the limits so i took the comment with the line

File: pallets/system/src/header_builder.rs (hunk 2 above)
Comment (Leouarz) [reply pallets/system/src/header_builder.rs:19]: 💀

**Review by ToufeeqP — APPROVED** [review]

**Pull request closed without merging**